### ISSUE TITLE
ImportError: /lib64/libc.so.6: version `GLIBC_2.14'  not found

### ISSUE THREAD

**Issue by jenchien** [kundajelab/atac_dnase_pipelines#33]
Hi kundajelab,

Thank you for making the atac pipelines available. I recently installed the pipeline successfully and was testing our human data. Alignment, peak calling, and idr were done, but it ran into fatal error during ataqc:

> ImportError: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /home/chang-j/anaconda2/envs/bds_atac/lib/python2.7/site-packages/pysam/libchtslib.so)

I noticed the most recent commit `ae8a65c` is related to glibc, so I tried` git pull` this version,  `./uninstall_dependencies.sh`,  `./install_dependencies.sh` and resume the run. However the same error continues to show up. I attached the full stdout here. If you need more info to identify the problem please kindly let me know. Thank you in advance for help.

[error-atac.bds.txt](https://github.com/kundajelab/atac_dnase_pipelines/files/753745/error-atac.bds.txt)

**Comment by leepc12** [comment]
This GLIBC issue seems to happen on old linux sytems. The commit ae8a65c (removing GLIBC from the installer) is to revert the commit fa55714 (adding GLIBC to the installer). Including GLIBC in the installer caused another issue #32 . I am still figuring this out.

Can you try the following (manually installing GLIBC)?

$ source activate bds_atac
$ conda install glibc -c asmeurer
$ source deactivate

**Comment by jenchien** [comment]
Thank you for the prompt reply. I tried your suggestion for manually installing GLIBC. The installation seems successful, but the resuming run encountered another fatal error like attached.

[error-atac.bds.after_manual_glibc.txt](https://github.com/kundajelab/atac_dnase_pipelines/files/754014/error-atac.bds.after_manual_glibc.txt)

**Comment by rleylek** [comment]
I'm having the same problem, following.

**Comment by jchap14** [comment]
Also, experiencing this ``GLIBC_2.14' not found` issue on SCG4. Thanks Jin

**Comment by leepc12** [comment]
I am still figuring this out. This worked on SCG4.

$ source activate bds_atac
$ conda install pysam
$ conda uninstall glibc
$ source deactivate

Please let me know if this works.

**Comment by jchap14** [comment]
$ conda uninstall glibc

Fetching package metadata .........
Solving package specifications: .

PackageNotFoundError: Package not found: '' no packages found to remove from environment: /home/jchap14/miniconda3/envs/bds_atac

**Comment by rleylek** [comment]
Yup, I get the same on Sherlock. And for installing pysam it says "packages already installed"

**Comment by rleylek** [comment]
Oh, looks like glibc is commented out of the requirements.txt...?

**Comment by jenchien** [comment]
Installing pysam and uninstalling glibc went through on my side, but the same error (GLIBC_2.14' not found) occurs when running the pipeline.

**Comment by leepc12** [comment]
Okay, I think I fixed it (commit 75d578d). I tested on SCG4 and Sherlock.
Please git pull the pipeline code first, run `./uninstall_dependencies.sh` and then `./install_dependencies`.

**Comment by jenchien** [comment]
This does fix the glibc issue. Thank you so much!

**Comment by sriharsha-sammeta** [comment]
`conda install tensorflow` works

**Comment by rogeroyer** [comment]
I have been meeting this problem yet, but I have solved it after upgrading "glibc2.12" to "glibc2.17". Here is a solution in CSDN, wish it can help you.
- https://blog.csdn.net/wyl9527/article/details/78256066

**Comment by R-Gaurav** [comment]
Just leaving it here in case someone finds it useful:

I installed ray 0.8.7 on `CentOS Linux release 7.9.2009 (Core)` with pyhon 3.7.7 environment and after importing ray, it complained of `Original error was: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /home/rgaurav/miniconda3/envs/latest-nengo-tf/lib/python3.7/site-packages/numpy/core/_multiarray_umath.cpython-37m-x86_64-linux-gnu.so)`. It was prompted that NumPy vesion is 1.21.0. Therefore, I downgraded the NumPy version to 1.19.2 and then the error was fixed.
